SayTheirNames/Source/Model/Person.swift

```diff
@@ -1,28 +1,63 @@
 //
 //  Person.swift
-//  SayTheirNames
+//  Say Their Names
 //
 //  Created by Ahmad Karkouti on 30/05/2020.
 //  Copyright © 2020 Franck-Stephane Ndame Mpouli. All rights reserved.
 //
 
 import Foundation
 
-struct Person: Codable {
-    var id: String
+public struct Person: Decodable {
+    var id: Int
     var fullName: String
-    var age: Int
-    var childrenCount: Int
-    var date: Date
-    var location: String
-    var media: [String]
+    var dob: String
+    var doi: String
+    var childrenCount: String
+    var age: String
+    var city: String
+    var country: String
     var bio: String
     var context: String
-    var donations: [Donation]
-    var petitions: [Petition]
+    var images: [Image]
+    var donations: Donations?
+    var petitions: Petitions?
+    var media: [Media]?
+    var socialMedia: [SocialMedia]?
```

Comment:
do we need an optional? could array just be empty, or is it because of the way `Codable` works?

Reply:
I can make it empty - it was choice.

Reply:
my thinking is that if there's no meaningful difference between nil and [] (i.e. both mean that there's no media) - then it's easier to handle at usage site

Reply:
I think handling `nil` at the UI level would be best. This way we don't have floating instance and /wasting/ memory for a simple check.

Reply:
at the usage site you'd have to do
```
if person?.socialMedia.isEmpty == false ?? false {
   // hide
}
```
or
```
if let sm = person?.socialMedia, !sm.isEmpty {
   // hide
}
```
because we don't want to show the section if it's an empty array. 

With non-optional check it's simply an emptiness check. Empty array almost doesn't take up any memory, and nil is a case of an `Optional` enum which also takes memory.

Reply:
The backend will not provide an empty array for a given key - thus at parsing time it will not return an empty array.

That said the check would simply be:

`if let sm = person?.socialMedia {
   // hide/show section
}`

I agree but the memory use for an `Array` vs `nil` is a vast difference.

Reply:
I would agree, if we could rely on the backend always sending non-empty arrays. On the client side we have to account for all cases though - if there comes null, wrong type, etc - the app should work gracefully and not crash / show buggy UI.

We can still check it as-is, just more characters. Up to you, anyway, it's a minor thing 😊

Reply:
I added a `stud` for each, so we no longer have `Optional`s.